modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2177,6 +2177,17 @@ public CodegenProperty fromProperty(String name, Schema p) {
             //    property.baseType = getSimpleRef(p.get$ref());
             //}
             // --END of revision
+
+            // set is enum to true when a model is enum and type is non-primitive.
+            if (StringUtils.isNotBlank(p.get$ref()) && globalSchemas != null) {
+               Schema ref = globalSchemas.get(ModelUtils.getSimpleRef(p.get$ref()));
+               if (ref != null) {
+                   if (ref.getEnum() != null) {
```

Comment:
@zhemant What about the following instead?
```
if (ref != null && ref.getEnum() != null) {
...
}
```

Reply:
Yes, this is better. I will make these changes and push again.